### ISSUE TITLE
Question and recommendation: YOLOv4-tiny transfer learning for persons indoor detection

### ISSUE THREAD

**Issue by KyryloAntoshyn** [AlexeyAB/darknet#6639]
Hi, @AlexeyAB, I have some questions about my task: persons detection on Jetson Nano. For this task I will use the camera that will be ~3 meters above the floor and my system needs to detect persons (after it I will use this data inside another system). System will be deployed in the mall (indoor). I've read the documentation about how to train already pre-trained model (perform transfer learning) to detect my custom objects and have some questions:

1. YOLOv4-tiny COCO has been already trained. I know that COCO dataset contains Person class. **Will the quality of persons detection be better if I make transfer learning of this pre-trained model on another dataset with people, for example, Caltech Pedestrian Dataset, or it is better to use this model as it is?**

2. If the answer of the first question is "Yes", **which datasets could you recommend for persons detection that can be suitable for my task (persons detection in the mall)?** I found Google Open Images dataset also.

3. **Is it better to use ready-made datasets for training or I need to make pictures at the location where my system will work** and annotate them myself (maybe use pseudo-labeling)?

Thank you in advance!

**Comment by AlexeyAB** [comment]
1. Usually better to train on your own dataset, by using pre-trained MSCOCO weights
2. You can try to combine MSCOCO+BDD+OpenImages or try to find something more suitable there: https://academictorrents.com/browse.php
3. Its better to collect your own 2000 - 10 000 labeled images from the same cameras where it will be used

**Comment by KyryloAntoshyn** [comment]
Thank you so much for your help!

**Comment by stephanecharette** [comment]
@KyryloAntoshyn you'll probably need to go with AlexeyAB's 3rd option.  See this example I did for a customer earlier this year:  https://www.ccoderun.ca/programming/ml/people_counter.html

Will your camera be pointing out, or 90 degrees down?  I found no data set with cameras pointing 90 degrees down.  It of course will always be better when trained with images captured at the actual location.  Remember that summer and winter will result in different images due to hats, coats, etc, so be ready to re-train in 6 months with a bunch of new pictures to top up the neural network.

And remember to include pictures without people in them!  "negative" images.  So the network doesn't start finding faces in the swirl patterns on the floor, etc.

**Comment by KyryloAntoshyn** [comment]
@stephanecharette thanks for your consultation! The idea is to find person's coordinate in a confined space. The system must work outdoors too so that we will setup cameras diagonally to the floor. I want to compare 2 approaches: using bottom middle bounding box point (object detection) or middle point between two legs (pose estimation), from the camera perspective in relation to the floor. This will be used in another system.

**Comment by KyryloAntoshyn** [comment]
**Hi, @AlexeyAB! I've performed my first "test" transfer learning on 3 classes: Person, Human leg and Human foot from Google Open Images Dataset. I was following your guidelines in the README. They are pretty straightforward, you did an excellent work, thanks!**

**My dataset has 2000 random training and 400 random validation images for class "Person".** Moreover, I have some training and validation images for classes "Human leg" and "Human foot", but less then 2000 and 400 respectively (this is because I was downloading images with flags: IsTruncated=0, IsGroupOf=0, IsDepiction=0 and IsInside=0, which helps to get more accurate images, but reduces the number of appropriate images).

**I have some questions about training:**
1. After training there is _yolov4-tiny-OIDv4-Person-HumanLeg-HumanFoot_best.weights_ file inside the backup directory. I ran the "map" command in order to calculate mAP value and had following results: Person – 53.18 %, Human leg – 34.25 % and Human foot – 1.50 %. These results are shown on the image below.
![image](https://user-images.githubusercontent.com/26432700/93720980-9b27d800-fb95-11ea-9507-14e17edaa60e.png)
**What formula do you use for mAP calculation?** When I try to use _mAP = TP / (TP + FP)_ formula I get different results: _270 / (270 + 183) = ~0.6_ for class Person.

2. My training is actually not good, because I have only 53.18 % mAP for class Person (as shown on the image above) and my model detects persons worse then pre-trained on the COCO dataset model (confidences are less, bounding boxes are not so close to the objects). **Is this due to small number of training images or there can be a lot of other reasons: lack of "negative" images, different images in training/validation datasets by shape, side of object, relative size, angle of rotation, tilt, illumination?**

3. I want to train the YOLOv4-tiny model for Pedestrian detection by combining COCO, BDD, Open Images as you recommended above. **How many images do I need to consider in order to train the model successfully: 2000, 10000 or more? Negative images number must be equal to the number of images with persons, right? Do I need to use images from the COCO dataset despite the fact that the model is pre-trained on this dataset? Is it enough to have GTX 1050 for this task (my test training took less then 4 hours for my model)?**

4. **As I understand, I can use trained on COCO model in order detect only "Person" class, but in this case I need to filter detected objects in code.** I can't adjust the _yolov4-tiny.cfg_ file (change the number of classes and filters in [yolo] layers), because it won't suit the trained model. **Is it correct?**

5. **How the quality of training differs from subdivisions = 16, 32 or 64?** For _yolov4-tiny-custom.cfg_ subdivisions=1. **Do I need to change it to 16 as stated in the README "How to train" section?**

6. **"Out of memory" error can occur due to small subdivisions number, because GPU doesn't have enough memory to process a lot of mini_batch samples at once, right?**

7. Repo contains _yolov4-tiny-3l.cfg_ model. **This model better detects small objects due to 3 [yolo] layers сompared with yolov4-tiny.cfg, right?**
 
8. On the graph I can see "C" value. **Is it contrastive loss?**
![image](https://user-images.githubusercontent.com/26432700/93721298-c3183b00-fb97-11ea-93e1-97cb0e110e8e.png)

**Thank you in advance!**
